lib/tasks/config.rake

```diff
@@ -0,0 +1,111 @@
+    puts " List appliances                    - Usage: rake config:list_appliances"
+    puts " List zones                         - Usage: rake config:list_zones"
+  end
+
+  desc "List all roles available"
+  task :list_roles do
+    puts role_list.to_json
+  end
+
+  desc "List appliance active roles"
+  task :list_active_roles => [:environment] do
+    if ENV['APPLIANCE_ID'].blank?
+      puts MiqServer.my_server.active_role_names.to_json
+    else
+      puts MiqServer.find(ENV['APPLIANCE_ID']).role.split(",").to_json
+    end
+  end
+
+  desc "Set appliance roles"
+  task :set_roles => [:environment] do
+    begin
+      raise "You must specify a valid list of server foles" if ENV['SERVER_ROLES'].blank?
+      server_roles = JSON.parse(ENV['SERVER_ROLES'])
+      server_roles.each do |role|
+        raise "Role #{role} not available" unless role_list.include? role
+      end
+      if ENV['APPLIANCE_ID'].blank?
+        MiqServer.my_server.set_config(:server => {:role => server_roles.join(",")})
+      else
+        MiqServer.find(ENV['APPLIANCE_ID']).set_config(:server => {:role => server_roles.join(",")})
+      end
+    rescue => err
+      STDERR.puts err.message
+      exit(1)
+    end
+  end
+
+  desc "Get appliance config summary"
+  task :summary => [:environment] do
+    puts "Appliance name: #{MiqServer.my_server.name}"
+    puts "Region number: #{MiqServer.my_server.region_number}"
+    puts "Region: #{MiqServer.my_server.region_description}"
+    puts "Zone ID: #{MiqServer.my_server.zone_id}"
+    puts "Zone: #{MiqServer.my_server.zone_description}"
+    puts "Active roles: #{MiqServer.my_server.active_role}"
+  end
+
+  desc "Create zone"
+  task :create_zone => [:environment] do
+    begin
+      raise "You must specify zone name and zone description" if ENV['NAME'].blank? || ENV['DESCRIPTION'].blank?
+      zone = Zone.new({'name': ENV['NAME'], 'description': ENV['DESCRIPTION']})
+      zone.save
+    rescue => err
+      STDERR.puts err.message
+      exit(1)
+    end
+  end
+
+  desc "Modify appliance zone"
+  task :modify_zone => [:environment] do
+    begin
+      raise "You must specify a existent zone name and appliance name" if ENV['APPLIANCE_ID'].blank? || ENV['ZONE_NAME'].blank?
+      appliance = MiqServer.find(ENV['APPLIANCE_ID'])
+      zone = Zone.find_by(name: ENV['ZONE_NAME'])
+      if zone.nil? || appliance.nil?
+        raise "Zone or appliance not found"
+      end
+      appliance.zone = zone
+      appliance.save
+      config = VMDB::Config.new("vmdb")
+      config.config[:server][:zone] = ENV['ZONE_NAME']
+      config.validate
+      config.save
+    rescue => err
+      STDERR.puts err.message
+      exit(1)
+    end
+  end
+
+  desc "List available appliances"
+  task :list_appliances=> [:environment] do
+    arr = []
+    MiqServer.all.as_json.each do |s|
+      arr << s.slice("id", "name")
```

Comment:
`rake evm:status_full` ?  Maybe we can make that rake task limit what columns comes back?